script/wait_for_file.sh

```diff
@@ -0,0 +1,8 @@
+wait_for_file() {
+  local file="$1"; shift
+  local wait_seconds="${1:-10}"; shift # 10 seconds as default timeout
+
+  until test $((wait_seconds--)) -eq 0 -o -f "$file" ; do sleep 1; done
+
+  ((++wait_seconds))
```

Comment:
Ah, this is intended as the return value. However, in the other script you use it as `wait_for_file ... || error`. Your return code will always be non-zero, since you increment from 0 or larger integer.

Maybe last statement (return value) should be `[ "$wait_seconds" -ne 0 ]`?